test-utils/src/main/java/org/hibernate/validator/testutil/ConstraintViolationAssert.java

```diff
@@ -340,12 +340,12 @@ public static PathExpectation pathWith() {
 		return new PathExpectation();
 	}
 
-	public static class ConstraintViolationSetAssert extends CollectionAssert {
+	public static class ConstraintViolationSetAssert extends ListAssert {
 
 		private final Set<? extends ConstraintViolation<?>> actualViolations;
 
 		protected ConstraintViolationSetAssert(Set<? extends ConstraintViolation<?>> actualViolations) {
-			super( actualViolations );
+			super( new ArrayList(actualViolations) );
```

Comment:
Have you checked if IterableAssert could be sufficient as a base class? It would avoid this conversion.